key-attestation/java-example/src/main/java/com/fortanix/keyattestationstatementverifier/certchecker/KeyAttestationStatementCertChecker.java

```diff
@@ -0,0 +1,113 @@
+import java.security.interfaces.RSAPublicKey;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.cert.X509CertificateHolder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateHolder;
+
+import com.fortanix.keyattestationstatementverifier.Common;
+import com.fortanix.keyattestationstatementverifier.KeyAttestationStatementVerifyException;
+import com.fortanix.keyattestationstatementverifier.types.asn1.KeyUsage;
+
+public class KeyAttestationStatementCertChecker extends CertChecker {
+    private static final Logger LOGGER = Logger.getLogger(KeyAttestationStatementCertChecker.class.getName());
+    private String certCN;
+
+    @Override
+    public void check(X509Certificate cert, X509Certificate issuerCert) throws Exception {
+        LOGGER.info("Checking certificate content:\n" + cert.toString());
+        certCN = Common.getCommonName(cert);
+        LOGGER.info(String.format(
+                "Checking '%s' certificate's Validity", certCN));
+        cert.checkValidity();
+        LOGGER.info(String.format(
+                "Checking '%s' certificate's Subject & Issuer", certCN));
+        check_subject_and_issuer(cert);
+        LOGGER.info(String.format(
+                "Checking '%s' certificate's Public key length & type", certCN));
+        check_public_key(cert);
+        LOGGER.info(String.format(
+                "Checking '%s' certificate's extensions", certCN));
+        check_extensions(cert, issuerCert);
+    }
+
+    private void check_subject_and_issuer(X509Certificate cert) throws Exception {
+        X509CertificateHolder statementCert = new JcaX509CertificateHolder(cert);
+        // check subject
+        X500Name statementCertSubject = statementCert.getSubject();
+        if (!Common.checkNameMatch(statementCertSubject, Common.KEY_ATTESTATION_STATEMENT_NAME)) {
+            throw new KeyAttestationStatementVerifyException(
+                    Common.KEY_ATTESTATION_STATEMENT_CN + " certificate subject is invalid " +
+                            statementCertSubject.toString());
+        }
+        // check issuer
+
+        X500Name statementCertIssuer = statementCert.getIssuer();
+        if (!Common.checkNameMatch(statementCertIssuer, Common.DSM_CLUSTER_KEY_ATTESTATION_AUTHORITY_NAME)) {
+            throw new KeyAttestationStatementVerifyException(
+                    Common.KEY_ATTESTATION_STATEMENT_CN + " certificate issuer is invalid "
+                            + statementCertIssuer.toString());
+        }
+    }
+
+    private void check_public_key(X509Certificate cert) throws Exception {
+        PublicKey statementPk = cert.getPublicKey();
+        if (statementPk instanceof RSAPublicKey) {
+            RSAPublicKey statementRsaPk = (RSAPublicKey) statementPk;
+            assert (statementRsaPk.getModulus().bitLength() >= 2048);
```

Comment:
And even if the keys _are_ RSA, what makes you sure that they are 2K?

Reply:
You are right, since we cannot expect what kind of asymmetric key will be, I think we can skip the check of public key here.

Reply:
resolving since you removed the check